### PR TITLE
feat(List): add IconTextCell column

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
On the same list as the PR before, we'll have to get some cells with icons then a value
image

**What is the chosen solution to this problem?**
Add a new column renderer

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/VirtualizedList/CellIconText/IconTextColumn.component.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import CellIconText from './CellIconText.component';
+
+export const cellType = 'icontext';
+export const iconTextColumnConfiguration = {
+	cellRenderer: props => <CellIconText {...props} />,
+};
+
+// this is a fake component to be usable in JSX,
+// but the element is used as props object internally (VirtualizedList / RV)
+export default function BooleanColumn() {
```
Comment (ncomont) [inline packages/components/src/VirtualizedList/CellIconText/IconTextColumn.component.js:12]: Boolean ? 😁

Comment (romainseb) [reply packages/components/src/VirtualizedList/CellIconText/IconTextColumn.component.js:12]: https://gph.is/2cg7pt1

File: packages/components/src/VirtualizedList/CellIconText/CellIconText.component.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		return this.props.cellData !== nextProps.cellData;
+	}
+
+	render() {
+		const { cellData } = this.props;
+
+		return (
+			<div className={css('tc-virtualizedlist-cell-icon-text')}>
+				<Icon name={cellData.icon} />
+				<span className={css('tc-virtualizedlist-cell-icon-text-label')}>{cellData.label}</span>
```
Comment (jsomsanith-tlnd) [inline packages/components/src/VirtualizedList/CellIconText/CellIconText.component.js:23]: with css modules, this classname will be something like `tc-virtualizedlist-cell-icon-text__tc-virtualizedlist-cell-icon-text-label__hash`

I think you should have a sass like this
```
.tc-virtualizedlist-cell-icon-text {
    .label {}
}
```
And in the js code 
```
<span className={css.label}>
```

File: packages/components/src/VirtualizedList/CellIconText/CellIconText.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.tc-virtualizedlist-cell-icon-text {
+	display: flex;
+	:global(.tc-svg-icon) {
+		margin-right: $padding-small;
```
Comment (jsomsanith-tlnd) [inline packages/components/src/VirtualizedList/CellIconText/CellIconText.scss:4]: If you set a margin left to the label, it avoids the global

File: packages/components/src/VirtualizedList/CellIconText/__snapshots__/CellIconText.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`CellBoolean should render an empty cell 1`] = `
+<div
+  className="tc-virtualizedlist-cell-icon-text theme-tc-virtualizedlist-cell-icon-text"
+>
+  <Icon />
```
Comment (jsomsanith-tlnd) [inline packages/components/src/VirtualizedList/CellIconText/__snapshots__/CellIconText.test.js.snap:7]: In Icon component
```
if (!name) {
	return <div className="alert alert-danger">Icon: no name provided</div>;
}
```

Let's condition the icon rendering

File: packages/components/src/VirtualizedList/CellIconText/CellIconText.scss [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
-	}
+}
 
-	&-label {
-		min-width: 0;
-		text-overflow: ellipsis;
-		overflow: hidden;
-		white-space: nowrap;
-	}
+.label {
```
Comment (jsomsanith-tlnd) [inline packages/components/src/VirtualizedList/CellIconText/CellIconText.scss:5]: If you still want css module to output you a classname with context (~`tc-icon-text__label__hash`), you can nest `.label` in `.tc-icon-text`.
I approve anyway, as the label style won't leak in the current state, it's just a matter of your taste :)

Comment (romainseb) [reply packages/components/src/VirtualizedList/CellIconText/CellIconText.scss:5]: I'll give another try on that

**Pull request merged**